src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/Dispensers.cs

```diff
@@ -62,6 +59,41 @@ internal static RuntimeAssembly GetRuntimeAssemblyFromByteArray(byte[] rawAssemb
             return result;
         }
 
+        /// <summary>
+        /// Returns non-null or throws.
+        /// </summary>
+        internal static RuntimeAssembly GetRuntimeAssemblyFromPath(string assemblyPath, string nativeImagePath)
+        {
+            byte[] rawAssembly;
+            using (var stream = new FileStream(assemblyPath, FileMode.Open, FileAccess.Read, FileShare.Read | FileShare.Delete))
```

Comment:
Can this just map the file instead of reading it into byte array?

Reply:
You mean like here? https://github.com/dotnet/corert/blob/c142d456570b105d18bfb92229bddb78e7f4cfd8/src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs#L155-L167

Just gotta figure how to access `MemoryMappedFile` from `System.Private.Reflection.Core`

Reply:
It will be easier after https://github.com/dotnet/corert/pull/8015/files#diff-ed8f8c4ffb723b549cbacb77ae54430eR36 is merged.

Reply:
Hm after rebasing (and doing a `git clean -xfd`, `build.cmd clean`), VS no longer recognizes `ImmutableArray`, `MetadataReader`, `PEReader` when opening https://github.com/dotnet/corert/blob/b8b7cb6abc9175e13ce3ff11d65980774db65a06/src/System.Private.TypeLoader/src/Internal/Reflection/Execution/AssemblyBinderImplementation.Ecma.cs#L52-L54
from the `System.Private.TypeLoader.Experimental` project.

Reply:
Aaand i've run into the trap again that i must set Architecture back to x64 after killing everything  😞